src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts

```diff
@@ -8,35 +8,138 @@ import { createChatParticipant } from '../ChatParticipantUtils';
+import { orgChangeErrorEvent, orgChangeEvent } from '../../OrgChangeNotifier';
 
+export interface OrgDetails {
+    orgID: string;
+    orgUrl: string;
+}
 
 export class PowerPagesChatParticipant {
+    private static instance: PowerPagesChatParticipant | null = null;
     private chatParticipant: vscode.ChatParticipant;
     private telemetry: ITelemetry;
+    private extensionContext: vscode.ExtensionContext;
+    private readonly _pacWrapper?: PacWrapper;
+    private isOrgDetailsInitialized = false;
+    private readonly _disposables: vscode.Disposable[] = [];
+    private cachedEndpoint: { intelligenceEndpoint: string, geoName: string } | null = null;
+
+    private orgID: string | undefined;
+    private orgUrl: string | undefined;
 
-    constructor(context: vscode.ExtensionContext, telemetry: ITelemetry | TelemetryReporter,) {
+    private constructor(context: vscode.ExtensionContext, telemetry: ITelemetry | TelemetryReporter, pacWrapper?: PacWrapper) {
 
         this.chatParticipant = createChatParticipant('powerpages', this.handler);
 
         //TODO: Check the icon image
         this.chatParticipant.iconPath = vscode.Uri.joinPath(context.extensionUri, 'src', 'common', 'chat-participants', 'powerpages', 'assets', 'copilot.png');
 
         this.telemetry = telemetry;
+
+        this.extensionContext = context;
+
+        this._pacWrapper = pacWrapper;
+
+        this._disposables.push(orgChangeEvent(async (orgDetails: ActiveOrgOutput) => {
+            await this.handleOrgChangeSuccess(orgDetails);
+        }));
+
+        this._disposables.push(orgChangeErrorEvent(async () => {
+            await createAuthProfileExp(this._pacWrapper);
+        }));
+
+    }
+
+    public static getInstance(context: vscode.ExtensionContext, telemetry: ITelemetry | TelemetryReporter, pacWrapper?: PacWrapper) {
+        if (!PowerPagesChatParticipant.instance) {
+            PowerPagesChatParticipant.instance = new PowerPagesChatParticipant(context, telemetry, pacWrapper);
+        }
+
+        return PowerPagesChatParticipant.instance;
+    }
+
+    public dispose() {
+        this.chatParticipant.dispose();
     }
 
     private handler: vscode.ChatRequestHandler = async (
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _request: vscode.ChatRequest,
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        request: vscode.ChatRequest,
         _context: vscode.ChatContext,
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        _stream: vscode.ChatResponseStream,
+        stream: vscode.ChatResponseStream,
         // eslint-disable-next-line @typescript-eslint/no-unused-vars
         _token: vscode.CancellationToken
     ): Promise<IPowerPagesChatResult> => {
         // Handle chat requests here
 
-        // TODO: Handle authentication and org change
+        //TODO: Move strings to constant once finalized
+        stream.progress('Working on it...')
```

Comment:
dont defer this - this should be part of initial structure - we can just keep on updating the content later